packages/tools/src/utilities/segmentation/getUniqueSegmentIndices.ts

```diff
@@ -0,0 +1,74 @@
+import { Types, cache } from '@cornerstonejs/core';
+import { SegmentationRepresentations } from '../../enums';
+import { getSegmentation } from '../../stateManagement/segmentation/segmentationState';
+import { isVolumeSegmentation } from '../../tools/segmentation/strategies/utils/stackVolumeCheck';
+
+/**
+ * Retrieves the unique segment indices from a given segmentation.
+ *
+ * @param segmentationId - The ID of the segmentation.
+ * @returns An array of unique segment indices.
+ * @throws If no geometryIds are found for the segmentationId.
+ */
+function getUniqueSegmentIndices(segmentationId) {
+  const segmentation = getSegmentation(segmentationId);
+
+  if (segmentation.type === SegmentationRepresentations.Labelmap) {
+    const labelmapData =
+      segmentation.representationData[SegmentationRepresentations.Labelmap];
+
+    const keySet = {};
+
+    if (isVolumeSegmentation(labelmapData)) {
+      const volume = cache.getVolume(segmentationId);
+      const scalarData = volume.getScalarData();
+
+      for (let i = 0; i < scalarData.length; i++) {
+        const segmentIndex = scalarData[i];
+        if (segmentIndex !== 0 && !keySet[segmentIndex]) {
+          keySet[segmentIndex] = true;
+        }
+      }
+    } else {
+      labelmapData.imageIdReferenceMap.forEach((segmentationImageId) => {
+        const image = cache.getImage(segmentationImageId);
+        const scalarData = image.getPixelData();
+
+        for (let i = 0; i < scalarData.length; i++) {
+          const segmentIndex = scalarData[i];
+          if (segmentIndex !== 0 && !keySet[segmentIndex]) {
+            keySet[segmentIndex] = true;
+          }
+        }
+      });
+    }
+
+    return Object.keys(keySet).map((it) => parseInt(it, 10));
```

Comment:
The ,10 is implicit in the parseInt

Reply:
Also, the other response is sorted, but this one isn't - please sort this one or don't sort the others.